ci/codecov.sh

```diff
@@ -23,51 +23,69 @@ set -ex
 if [[ "$1" == "setup" ]]; then
     export B2_VARIANT=debug
     if [ -z "$B2_CI_VERSION" ]; then
-        export B2_CXXFLAGS="${B2_CXXFLAGS:+$B2_CXXFLAGS }cxxflags=-fkeep-static-functions cxxflags=--coverage"
+        export B2_CXXFLAGS="${B2_CXXFLAGS:+$B2_CXXFLAGS }cxxflags=-fprofile-arcs cxxflags=-ftest-coverage cxxflags=--coverage"
     else
-        export B2_CXXFLAGS="${B2_CXXFLAGS:+$B2_CXXFLAGS }-fkeep-static-functions --coverage"
+        export B2_CXXFLAGS="${B2_CXXFLAGS:+$B2_CXXFLAGS }-fprofile-arcs -ftest-coverage --coverage"
     fi
     export B2_LINKFLAGS="${B2_LINKFLAGS:+$B2_LINKFLAGS }--coverage"
 
 elif [[ "$1" == "upload" ]]; then
     if [ -z "$GCOV" ]; then
-        ver=7 # default
-        if [ "${B2_TOOLSET%%-*}" == "gcc" ]; then
-            if [[ "$B2_TOOLSET" =~ gcc- ]]; then
-                ver="${B2_TOOLSET##*gcc-}"
-            elif [[ "$B2_COMPILER" =~ gcc- ]]; then
-                ver="${B2_COMPILER##*gcc-}"
+        # enforce.sh ensures B2_COMPILER
+        COMPILER=${B2_COMPILER}
+        COMPILER_FAMILY=$(echo ${COMPILER} | cut -d'-' -f1)
+        COMPILER_EDITION=$(echo ${COMPILER} | cut -d'-' -f2)
```

Comment:
Why don't you use bash variable substitution as done above?
```suggestion
        COMPILER_FAMILY="${COMPILER%%-*}"
        COMPILER_EDITION="${COMPILER##*-}"
```

Anyway: I wouldn't change this for backward compat reasons. Not that the comment is wrong: enforce.sh doesn't do anything for neither  B2_TOOLSET nor B2_COMPILER and older CI scripts only set B2_TOOLSET
And common_install.sh has code to deduce B2_TOOLSET from B2_COMPILER but not the other way round. Contrary: It explicitely checks if B2_COMPILER is set, so it may be unset.

So IMO for backwards compat reason this needs to stay checking B2_TOOLSET and B2_COMPILER in this order. Also only one of them may have a version.

The failing "Compatibility CI" confirms that